app/models/brand.rb

```diff
@@ -182,6 +184,24 @@ def in_trial?
     !!subscription.try(:trial?)
   end
 
+  def delete_account
+    ActiveRecord::Base.transaction do
+      unsubscribe_users_from_newsletter
+      subscription.cancel_plan! if subscription.present?
+      destroy
+    end
+  end
+
+  def unsubscribe_users_from_newsletter
+    users.each do |user|
+      user.update!(email_subscription: false) if user.email_subscription
+    end
+  end
+
+  def restore_brand
+    restore(recursive: true)
```

Comment:
Careful with this. We don't want to resurrect old listen signals, response groups, and responses that were deleted by the users.